fl4health/clients/adaptive_drift_constraint_client.py

```diff
@@ -0,0 +1,214 @@
+        BasicClient that packs the criterion loss (i.e. loss without the penalty) for exchange with the server and
+        expects to receive an updated (or constant if non-adaptive) parameter for the loss weight.
+
+        Args:
+            data_path (Path): path to the data to be used to load the data for client-side training
+            metrics (Sequence[Metric]): Metrics to be computed based on the labels and predictions of the client model
+            device (torch.device): Device indicator for where to send the model, batches, labels etc. Often 'cpu' or
+                'cuda'
+            loss_meter_type (LossMeterType, optional): Type of meter used to track and compute the losses over
+                each batch. Defaults to LossMeterType.AVERAGE.
+            checkpointer (Optional[ClientCheckpointModule], optional): Checkpointer module defining when and how to
+                do checkpointing during client-side training. No checkpointing is done if not provided. Defaults to
+                None.
+            metrics_reporter (Optional[MetricsReporter], optional): A metrics reporter instance to record the metrics
+                during the execution. Defaults to an instance of MetricsReporter with default parameters.
+            progress_bar (bool): Whether or not to display a progress bar during client training and validation.
+                Uses tqdm. Defaults to False
+        """
+        super().__init__(
+            data_path=data_path,
+            metrics=metrics,
+            device=device,
+            loss_meter_type=loss_meter_type,
+            checkpointer=checkpointer,
+            metrics_reporter=metrics_reporter,
+            progress_bar=progress_bar,
+        )
+        # These are the tensors that will be used to compute the penalty loss
+        self.drift_penalty_tensors: List[torch.Tensor]
+        # Exchanger with packing to be able to exchange the weights and auxiliary information with the server for
+        # adaptation
+        self.parameter_exchanger: FullParameterExchangerWithPacking
+        # Weight on the penalty loss to be used in backprop. This is what might be adapted via server calculations
+        self.drift_penalty_weight: float
+        # This is the loss value to be sent back to the server on which adaptation decisions will be made.
+        self.loss_for_adaptation: float
+        # Function to compute the penalty loss.
+        self.penalty_loss_function = WeightDriftLoss(self.device)
+
+    def get_parameters(self, config: Config) -> NDArrays:
+        """
+        Packs the parameters and training loss into a single NDArrays to be sent to the server for aggregation. If
+        the client has not been initialized, this means the server is requesting parameters for initialization and
+        just the model parameters are sent. When using the FedAvgWithAdaptiveConstraint strategy. This should not
```

Comment:
nit: When using the FedAvgWithAdaptiveConstraint strategy, this should not (seems like you want a comma and not period here)